src/PostListingDisplay/components/PostListingDisplay.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+
+import InputField from './InputField';
+
+import '../styles/PostListingDisplay.css';
+
+/**Component class that displays the input fields for a user to post a listing */
+class PostListingDisplay extends Component {
+    constructor(props){
+        super(props);
+
+        this.state = {
+            title: '',
+            description: '',
+            numRooms: '',
+            numSingles: '',
+            singlePrice: '',
+            numShared: '',
+            sharedPrice: '',
+            numBathrooms: '',
+            leaseTypes: '',
+            startDate: '',
+            endDate: '',
+            listingPrice: ''
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.calculateListingPrice = this.calculateListingPrice.bind(this);
+    }
+
+    handleChange(event) {
+        const value = event.target.value;
+        const name = event.target.name;
+        this.setState({
+            [name]: value
+        })
+        this.calculateListingPrice();
+    }
+
+    calculateListingPrice(){
+        const numSingles = this.state.numSingles;
+        const singlePrice = this.state.singlePrice;
+        const numShared = this.state.numShared;
+        const sharedPrice = this.state.sharedPrice;
+        const listingPriceValue = numSingles * singlePrice + numShared * sharedPrice;
+        if (isNaN(listingPriceValue)) {
+            this.setState({
+                listingPrice:'0'
+            });
```

Comment:
Do we want to set listing price to 0 here, or should we be keeping it empty so that it won't let you post the form unless there is a valid listing price?

Reply:
Yeah we want it empty. Fixed